Pod/Classes/PINAnimatedImageManager.m

```diff
@@ -355,9 +425,14 @@ + (void)processAnimatedImage:(NSData *)animatedImageData
     
     //close the file handle
     PINLog(@"closing last file: %@", fileHandle);
-    [fileHandle seekToFileOffset:0];
-    [fileHandle writeData:[NSData dataWithBytes:&frameCountForFile length:sizeof(frameCountForFile)]];
-    [fileHandle closeFile];
+    @try {
+      [fileHandle seekToFileOffset:0];
+      [fileHandle writeData:[NSData dataWithBytes:&frameCountForFile length:sizeof(frameCountForFile)]];
+      [fileHandle closeFile];
+    } @catch (NSException *exception) {
+      NSError *frameCountError = [NSError errorWithDomain:kPINAnimatedImageErrorDomain code:PINAnimatedImageErrorFileWrite userInfo:exception];
+      HANDLE_PROCESSING_ERROR(frameCountError);
+    } @finally {}
```

Comment:
HANDLE_PROCESSING_ERROR will close the file if there's an error, also, NSFileHandle will auto close the file on dealloc.